### PR TITLE
Improved some project stats

### PR DESCRIPTION
**Pull request by realDragonium**
After the home_project view has been refreshed it should show the correct amount of 
- stars
- watchers
- views

Improved the way the results from the database are mapped in V1ApiService.
Removed a NPE from Apiv1Controller when a project/plugin from a user didnt have a version yet.

### CODE REVIEW

File: src/main/java/io/papermc/hangar/controller/Apiv1Controller.java [hunk 1]
```diff
@@ -216,7 +216,7 @@ private ArrayNode writeProjects(List<ProjectsTable> projectsTables) {
             projectObj.set("href", mapper.valueToTree(project.getOwnerName() + "/" + project.getSlug()));
             projectObj.set("members", writeMembers(members.getOrDefault(project.getId(), new ArrayList<>())));
             projectObj.set("channels", mapper.valueToTree( projectChannels.getOrDefault(project.getId(), new ArrayList<>())));
-            projectObj.set("recommended", writeVersion(recommendedVersions.get(project.getId()), project, recommendedVersionChannels.get(project.getId()), vTags.getOrDefault(recommendedVersions.get(project.getId()).getId(), new ArrayList<>())));
+            projectObj.set("recommended", writeVersion(recommendedVersions.get(project.getId()), project, recommendedVersionChannels.get(project.getId()), vTags.getOrDefault(recommendedVersions.getOrDefault(project.getId(), new ProjectVersionsTable()).getId(), new ArrayList<>())));
```
Comment (KennyTV) [inline src/main/java/io/papermc/hangar/controller/Apiv1Controller.java:219]: looks a bit like a hacky solution 👀

File: src/main/java/io/papermc/hangar/db/dao/ProjectViewDao.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+@Repository
+@RegisterBeanMapper(ProjectViewsTable.class)
+//TODO: decide where we do count it as a view and where we dont
+// For now i will add it to the ProjectService
+public interface ProjectViewDao {
+
+    @SqlUpdate("insert into project_views (day, project_id, views) values (current_date , :projectId, 1)" +
+            "on conflict (day, project_id) " +
+            "do update set views = project_views.views + 1 ")
+    //The db will automatically get the current_date from itself (if you dont provide any date)
```
Comment (KennyTV) [inline src/main/java/io/papermc/hangar/db/dao/ProjectViewDao.java:21]: Move the command above the annotation

**Review by KennyTV — COMMENTED** [review]
Minor thing in general; add a space between comment starts `//` and their first word please

**Comment by realDragonium** [comment]
> Minor thing in general; add a space between comment starts `//` and their first word please
Will do, is this is a convention thing? Or is there a reason for it?

**Comment by Machine-Maker** [comment]
Are the changes to `init.sql` really necessary? We used the Timestamped annotation on insert dao methods with the `:now` instead of it defaulting to current date. we should keep doing in that way.

**Comment by KennyTV** [comment]
Yeah just style

**Comment by KennyTV** [comment]
If I'm not mistaken, I don't actually see the viewcount/starcount/watchcount gotten and displayed anywhere in your changes?

**Comment by realDragonium** [comment]
Yea that's correct, it was already being requested through the query on home_projects, so I only had to let it display the correct data. https://i.imgur.com/DJmgw3I.png

**Review by KennyTV — APPROVED** [review]

**Pull request merged**